### DISCUSSION TITLE
How to structure Github-style open design collaboration?

### DISCUSSION THREAD

**Discussion by GBKS** [uxbox/uxbox#1030] [General]
I am working on an [icon set](https://bitcoinicons.com) and a [wallet UI kit](https://github.com/GBKS/bitcoin-wallet-ui-kit) for Bitcoin applications as part of my work in the [Bitcoin Design Community](https://bitcoin.designhttps://bitcoin.design). The idea is to create public, community-owned design resources that anyone can use and contribute to, but I still haven't found a great way to make this work. While Figma community and their duplication feature have become useful for sharing, they have drastically limited the free collaboration features recently. It really requires lot of manual work-arounds right now to collaborate in a setting where contributors are not part of a tight team. I am curious what everyone thinks would be a good feature set to work towards to make this possible. Here's what I see:
- Public shared libraries and files that anyone can duplicate and use
- Branch, pull request, review and merge features to manage file iteration
- Comments, tasks or other means to communicate around change requests, proposals, etc

How do you currently organize open design collaboration? What can Penpot do right now, what's in the works, and what should this look like in the future? I really love what Penpot is doing, and as an open-source project, I think it could address open-source needs better than commercial projects.

**Comment by elhombretecla** [comment]
Hi @GBKS 

First of all, congratulations on your work with the icon set and UI kit, they are awesome :slightly_smiling_face:

One of Penpot's main motivations is to be able to contribute as much as possible in promoting the Open Source Design community. It is certainly a very ambitious challenge but we believe that it is possible and necessary. As you mentioned, there are today several tools which are very useful to share designs but not so much to contribute. At penpot we are working on creating functionalities focused on sharing designs and graphic assets as well as giving feedback on them and being able to contribute with changes.

As an open source design tool in alpha state, we still have a way to go in perfecting and developing Penpot itself, but we certainly want to start with the active community as soon as possible.

As you mentioned, a design version control system where being able to commit changes is one of the most interesting options. We have ideas along the lines of what you are asking for. We are listening to the community to explore solutions and we will take into account participations like yours. I think it is a great idea to put in common in this thread the your ideas and suggestions.

Thank you!!!

**Reply by GBKS** [reply]
Thanks for the prompt reply. I'm really blown away with what you're doing with Penpot, can't wait to see what else you are planning. For an alpha version, you're bringing a lot to the table. If it is helpful, I can write more about what we've tried, what has worked and what hasn't worked for us for open design collaboration. Maybe gathering user stories from different people could help identify the right way to go about this.

**Reply by elhombretecla** [reply]
That would be great!!!

Feel free to share your thoughts and research on open design collaboration. Our main communication channels are Github and our mail info@penpot.app

Thank you very much for your effort!

**Reply by GBKS** [reply]
Great. This is a big conversation, so I am not sure I can capture everything. Here are some of our big pain points at the moment:

1. Communal ownership
We currently don't have a way to make design files "owned by everyone". Ideally, there could be a community account (or public organization) that is managed by a trusted group of maintainers, and activity should be completely transparent. Right now, some of us publish Figma community files under our personal accounts, and we don't know of any options to do this differently.

1. Public contribution
With real-time editing, live collaboration in design tools has become seamless within trusted teams. In open-source however, contributions are often asynchronous and come from only loosely tied groups of people, sometimes new people just come in and suggest an improvement. This requires a very different set of mechanisms for coming together on what a design should be like. Ideally, someone could branch a file, make some changes, write a rationale for them and submit them for review, and the owner(s) of the file could easily compare, discuss, merge, or reject. We have no process for this right now and just copy & paste as needed. I believe this is a huge hinderance to contribution. I am not sure whether this should be a straight-out copy of how Github works, maybe there's a way to rethink these mechanisms for how designers work and think.

That's it for now. Another thing I see is that there is no "manual" or consensus anywhere on how this should work. It seems like everyone is hacking together some rough solution that works well-enough for their situation. Would be cool if we there would be some standard that everyone knows and is happy with a few years down the road.

**Comment by sbddesign** [comment]
I support this suggestion. A git-style version control for designers would be incredibly useful.

**Comment by Fristi** [comment]
As a coder I am missing this feature, would be awesome to have it
